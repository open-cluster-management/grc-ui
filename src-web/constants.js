/* Copyright (c) 2020 Red Hat, Inc. */

export default {
  EMPTY_CHOICE: 'EMPTY_CHOICE',
  overviewTopInfoClustersStr: 'string.clusters',
  overviewTopInfoAppStr: 'string.applications'
}
