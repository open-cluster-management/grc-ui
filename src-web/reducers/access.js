/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
import {
  USER_ACCESS_SUCCESS,
  USER_ACCESS_FAILURE,
  REQUEST_STATUS
} from '../actions'

export const access = (state = null, action) => {
  if (action) {
    switch (action.type) {
    case USER_ACCESS_SUCCESS:
      return { ...action, status: REQUEST_STATUS.DONE }
    case USER_ACCESS_FAILURE:
      return { ...state,
        status: REQUEST_STATUS.ERROR,
        errorMessage: action.err && action.err.details,
        statusCode: action.err && action.err.statusCode }
    default:
      return state
    }
  }
  return undefined
}
