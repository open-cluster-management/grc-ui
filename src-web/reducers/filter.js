/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
import {
  REQUEST_STATUS, RESOURCE_FILTERS_RECEIVE_SUCCESS, RESOURCE_FILTERS_UPDATE
} from '../actions'

export const resourceFilters = (state = {}, action) => {
  if (action) {
    switch (action.type) {
    case RESOURCE_FILTERS_RECEIVE_SUCCESS:
      return Object.assign({}, state, {
        status: REQUEST_STATUS.DONE,
        filters: {
          // disable cluster name filtering until server side fixes the issue
          // https://github.ibm.com/IBMPrivateCloud/roadmap/issues/10981
          // clusterNames: action.filters.clusterNames,
          clusterLabels: action.filters.clusterLabels,
        },
      })
    case RESOURCE_FILTERS_UPDATE: {
      const selectedFilters = state.selectedFilters ? {...state.selectedFilters} : {}
      selectedFilters[action.resourceName] = action.selectedFilters
      return {...state, selectedFilters}
    }
    default:
      return state
    }
  }
}
