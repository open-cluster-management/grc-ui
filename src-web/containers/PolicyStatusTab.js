/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import { Query } from 'react-apollo'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
// import resources from '../../lib/shared/resources'
// import { getResourceData } from '../tableDefinitions'
// import { connect } from 'react-redux'
// import { updateSecondaryHeader } from '../actions/common'
import { POLICY_STATUS } from '../../lib/client/queries'
import { Spinner } from '@patternfly/react-core'
import { DangerNotification } from '../components/common/DangerNotification'
import PolicyStatusView from '../components/modules/PolicyStatusView'
import { setRefreshControl } from '../../lib/client/reactiveVars'
// import { getTabs } from '../../lib/client/resource-helper'
import msgs from '../../nls/platform.properties'
import { LocaleContext } from '../components/common/LocaleContext'
import NoResource from '../components/common/NoResource'
import { AcmButton, AcmPage, AcmPageHeader, AcmAutoRefreshSelect, AcmRefreshTime, AcmSecondaryNav, AcmSecondaryNavItem } from '@open-cluster-management/ui-components'
import { INITIAL_REFRESH_TIME, REFRESH_INTERVALS, REFRESH_INTERVAL_COOKIE } from '../../lib/shared/constants'
import config from '../../lib/shared/config'

class PolicyDetailsTab extends React.Component{
  constructor(props) {
    super(props)
  }

  static contextType = LocaleContext

  render() {
    const {
      tab,
      policyName,
      policyNamespace: hubNamespace,
      location,
      history,
    } = this.props
    const { locale } = this.context
    const pollInterval = localStorage.getItem(REFRESH_INTERVAL_COOKIE) || INITIAL_REFRESH_TIME*1000
    return (
      <Query
        query={POLICY_STATUS}
        variables={{policyName, hubNamespace}}
        pollInterval={pollInterval}
        notifyOnNetworkStatusChange
      >
      {(result) => {
        const {data={}, loading, startPolling, stopPolling, refetch} = result
        const { status } = data
        if (!loading) {
          this.timestamp = new Date().toString()
        }
        setRefreshControl(loading, this.timestamp, startPolling, stopPolling, refetch)
        const error = status ? null : result.error
        if (error) {
          return (
            <DangerNotification error={error} />
          )
        } else if (loading && status === undefined) {
          return <Spinner className='patternfly-spinner' />
        } else if (Array.isArray(status) && status.length === 0) {
          return <NoResource
            title={msgs.get('no-status.title', [msgs.get('routes.grc', locale)], locale)}
            svgName='EmptyPagePlanet-illus.png'>
          </NoResource>
        } else {
            return (
              <AcmPage>
                <AcmPageHeader
                  title={policyName}
                  breadcrumb={[{ text: msgs.get('routes.policies', locale), to: config.contextPath }, { text: policyName, to: location.pathname}]}
                  controls={
                    <React.Fragment>
                      <AcmAutoRefreshSelect refetch={refetch}
                        refreshIntervals={REFRESH_INTERVALS}
                        refreshIntervalCookie={REFRESH_INTERVAL_COOKIE}
                        initRefreshTime={INITIAL_REFRESH_TIME} />
                      <AcmRefreshTime timestamp={this.timestamp} reloading={loading} />
                      <AcmButton id='edit-policy' isDisabled={false}
                        tooltip={msgs.get('error.permission.disabled', locale)}
                        onClick={() => history.push(`${config.contextPath}/edit`)}>
                        {msgs.get('routes.edit.policy', locale)}
                      </AcmButton>
                    </React.Fragment>
                  }
                  navigation={
                    <AcmSecondaryNav>
                      <AcmSecondaryNavItem
                        isActive={tab !== 'status'}
                        onClick={() => history.push(`${config.contextPath}/all/${hubNamespace}/${policyName}`)}>
                          {msgs.get('tabs.details', locale)}
                      </AcmSecondaryNavItem>
                      <AcmSecondaryNavItem
                        isActive={tab === 'status'}
                        onClick={() => history.push(`${config.contextPath}/all/${hubNamespace}/${policyName}/status`)}>
                          {msgs.get('tabs.status', locale)}
                      </AcmSecondaryNavItem>
                      {/* <AcmSecondaryNavItem
                        isActive={tab === 'yaml'}
                        onClick={() => history.push(`${config.contextPath}/all/${hubNamespace}/${policyName}/yaml`)}>
                          {msgs.get('tabs.yaml', locale)}
                      </AcmSecondaryNavItem> */}
                    </AcmSecondaryNav>
                  }>
                </AcmPageHeader>
                <PolicyStatusView
                  status={status}
                />
              </AcmPage>
            )
          }
        }}
      </Query>
    )
  }

}

PolicyDetailsTab.propTypes = {
  history: PropTypes.object,
  location: PropTypes.object,
  policyName: PropTypes.string,
  policyNamespace: PropTypes.string,
  tab: PropTypes.string,
}

export default withRouter(PolicyDetailsTab)
