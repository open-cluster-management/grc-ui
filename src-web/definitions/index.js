/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import lodash from 'lodash'

import { RESOURCE_TYPES } from '../../lib/shared/constants'

import hcmpoliciescluster from './hcm-policies-cluster'
import hcmpoliciespolicy from './hcm-policies-policy'
import hcmcompliances from './hcm-compliances'
import hcmpolicies from './hcm-policies'
import hcmsecurityfindings from './hcm-security-findings'
import hcmclusterfindings from './hcm-cluster-findings'
import hcmpoliciesapplication from './hcm-policies-application'

const resourceData = {
  [RESOURCE_TYPES.HCM_POLICIES_PER_APPLICATION.name]: hcmpoliciesapplication,
  [RESOURCE_TYPES.HCM_POLICIES_PER_POLICY.name]: hcmpoliciespolicy,
  [RESOURCE_TYPES.HCM_POLICIES_PER_CLUSTER.name]: hcmpoliciescluster,
  [RESOURCE_TYPES.HCM_COMPLIANCES.name]: hcmcompliances,
  [RESOURCE_TYPES.HCM_POLICIES.name]: hcmpolicies,
  [RESOURCE_TYPES.HCM_SECURITY_FINDINGS.name]: hcmsecurityfindings,
  [RESOURCE_TYPES.HCM_CLUSTER_FINDINGS.name]: hcmclusterfindings,
}

// diagram tabs
function getResourceData(resourceType) {
  // main Topology tab
  const def = resourceData[resourceType.name]
  if (!def) {
    //eslint-disable-next-line no-console
    console.error(`No resource data found for '${JSON.stringify(resourceType)}'`)
  }
  return def
}

export default getResourceData

export function getPrimaryKey(resourceType) {
  let pk = 'metadata.uid'
  const def = getResourceData(resourceType)
  if (def && def.primaryKey) {
    pk = def.primaryKey
  }
  return pk
}

export function getSecondaryKey(resourceType) {
  let sk = 'cluster'
  const def = getResourceData(resourceType)
  if(def && def.secondaryKey) {
    sk = def.secondaryKey
  }
  return sk
}

export function getURIKey(resourceType) {
  let uriKey = 'metadata.name'
  const def = getResourceData(resourceType)
  if (def && def.uriKey) {
    uriKey = def.uriKey
  }
  return uriKey
}

export function getDefaultSearchField(resourceType) {
  const def = getResourceData(resourceType)
  let sf = def && def.defaultSearchField
  if (!def || !def.tableKeys || (def && def.tableKeys.length < 1))
    //eslint-disable-next-line no-console
    console.error(`No table keys found in ${resourceType} resource definition`)
  if (!sf)
    sf = def && def.tableKeys && def.tableKeys[0] && def.tableKeys[0].resourceKey
  return sf
}

export function getDefaultSortField(resourceType) {
  const def = getResourceData(resourceType)
  let sf = def && def.defaultSortField
  if (!def || !def.tableKeys || (def && def.tableKeys.length < 1))
    //eslint-disable-next-line no-console
    console.error(`No table keys found in ${resourceType} resource definition`)
  if (!sf) {
    sf = def && def.tableKeys && def.tableKeys[0] && def.tableKeys[0].resourceKey
  }
  if (!sf)
    //eslint-disable-next-line no-console
    console.error(`No sortable fields defined for '${resourceType}' resource definition`)
  return sf
}

export function getTableKeys(resourceType) {
  const def = getResourceData(resourceType)
  return def.tableKeys
}

export function getLink(link, resource) {
  if(typeof(link) === 'boolean') {
    return `/${resource.metadata.namespace}/${resource.metadata.name}`
  } else if (typeof(link) === 'function') {
    return link(resource)
  } else {
    const parts = link.split('/')
    let path = ''
    parts.forEach(part => {
      const value = lodash.get(resource, part)
      path += `/${encodeURIComponent(value)}`
    })
    return path
  }
}
