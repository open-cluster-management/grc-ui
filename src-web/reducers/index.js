/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/*
 *
 * Things you should never do inside a reducer:
 *
 * - Mutate its arguments
 * - Perform side effects like API calls and routing transitions
 * - Call non-pure functions, e.g. Date.now() or Math.random()
 *
 * Reducers must be deterministic pure functions.  Given the same arguments, it should calculate the next state and return it.
 * No surprises. No side effects. No API calls. No mutations. Just a calculation.
 *
 * Selectors should sit along side reducers.
 */

import { createResourceReducer, resourceReducerFunction } from './common'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import lodash from 'lodash'

export { user, loggedIn } from './user'
export { uiconfig } from './uiconfig'
export { role } from './role'
export { userpreferences } from './userpreferences'
export { resourceToolbar } from './common'
export { secondaryHeader } from './common'

export { modal } from './modal'

export const HCMApplicationList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_APPLICATIONS))
export const HCMComplianceList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_COMPLIANCES))
export const HCMPolicyList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_POLICIES))
export const HCMPolicyApplicationList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_POLICIES_PER_APPLICATION))
export const HCMPolicyPolicyList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_POLICIES_PER_POLICY))
export const HCMPolicyClusterList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_POLICIES_PER_CLUSTER))
export const HCMSecurityFindingsList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_SECURITY_FINDINGS))
export const HCMClusterFindingsList = createResourceReducer(resourceReducerFunction, predicate.bind(null, RESOURCE_TYPES.HCM_CLUSTER_FINDINGS))

// export { default as catalog } from './catalog'
export { resourceFilters } from './filter'

export function predicate(resourceType, action) {
  if (lodash.isEqual(resourceType, action.resourceType)) {
    return true
  }
  const result = lodash.find(lodash.values(resourceType), type => {
    if (typeof type === 'string') {
      return type.indexOf(action.resourceType) > -1
    }
    return false
  })

  return result ? result : null
}
