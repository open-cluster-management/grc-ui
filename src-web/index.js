/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
import React from 'react'
import { render } from 'react-dom'
import { Provider } from 'react-redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import { BrowserRouter } from 'react-router-dom'
import { ApolloProvider } from 'react-apollo'
import { App } from './containers/App'
import * as reducers from './reducers'
import config from '../lib/shared/config'
import GrcApolloClient from '../lib/client/apollo-client'
import ScrollToTop from './components/common/ScrollToTop'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const loggerMiddleware = createLogger()
// Grab the state from a global variable injected into the server-generated HTML
const preloadedState = window.__PRELOADED_STATE__

const middleware = [thunkMiddleware] // lets us dispatch() functions
if (!window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ && config['featureFlags:reduxLogger']) {
  middleware.push(loggerMiddleware) // middleware that logs actions
}

// Allow the passed state to be garbage-collected
delete window.__PRELOADED_STATE__

// Create Redux store with initial state
const store = createStore(combineReducers(reducers), preloadedState, composeEnhancers(
  applyMiddleware(...middleware)
))



render(
  <ApolloProvider client={GrcApolloClient.getGrcClient()}>
    <Provider store={store}>
      <BrowserRouter>
        <ScrollToTop>
          <App />
        </ScrollToTop>
      </BrowserRouter>
    </Provider>
  </ApolloProvider>
  , document.getElementById('page')
)
