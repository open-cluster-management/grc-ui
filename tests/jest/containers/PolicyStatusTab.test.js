/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import * as reducers from '../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import GrcApolloClient from '../../../lib/client/apollo-client'
import { ApolloProvider } from 'react-apollo'
import { Provider } from 'react-redux'
import PolicyStatusTab from '../../../src-web/containers/PolicyStatusTab'

describe('PolicyStatusTab container test', () => {
  it('renders as expected with access', () => {
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const userAccess = [
      {
        namespace: 'default',
        rules: {
          '*/*': [
            '*'
          ]
        }
      }
    ]
    const component = renderer.create(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <Provider store={store}>
          <BrowserRouter>
            <PolicyStatusTab
              userAccess={userAccess}
            />
          </BrowserRouter>
        </Provider>
      </ApolloProvider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders as expected', () => {
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = renderer.create(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <Provider store={store}>
          <BrowserRouter>
            <PolicyStatusTab />
          </BrowserRouter>
        </Provider>
      </ApolloProvider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
