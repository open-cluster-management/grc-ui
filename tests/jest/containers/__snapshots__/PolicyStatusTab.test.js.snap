// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyDetail container test renders as expected 1`] = `
<div
  className="pf-c-page"
>
  <main
    className="pf-c-page__main"
    tabIndex={-1}
  >
    <section
      className="pf-c-page__main-section pf-m-light"
    >
      <div
        className="pf-l-split pf-m-gutter"
      >
        <div
          className="pf-l-split__item pf-m-fill"
        >
          <div
            className="pf-l-stack pf-m-gutter"
          >
            <div
              className="pf-l-stack__item"
            >
              <nav
                aria-label="Breadcrumb"
                className="pf-c-breadcrumb"
                data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
                data-ouia-component-type="PF4/Breadcrumb"
                data-ouia-safe={true}
              >
                <ol
                  className="pf-c-breadcrumb__list"
                >
                  <li
                    className="pf-c-breadcrumb__item"
                  >
                    <a
                      className="pf-c-breadcrumb__link"
                      href=""
                      onClick={[Function]}
                    >
                      Policies
                    </a>
                  </li>
                  <li
                    className="pf-c-breadcrumb__item"
                  >
                    <span
                      className="pf-c-breadcrumb__item-divider"
                    >
                      <svg
                        aria-hidden={true}
                        aria-labelledby={null}
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style={
                          Object {
                            "verticalAlign": "-0.125em",
                          }
                        }
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        />
                      </svg>
                    </span>
                    <a
                      aria-current="page"
                      className="pf-c-breadcrumb__link pf-m-current"
                    />
                  </li>
                </ol>
              </nav>
            </div>
            <div
              className="pf-l-stack__item"
            >
              <div
                className="pf-c-content"
              >
                <h1
                  className="pf-c-title pf-m-2xl"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="pf-l-split__item"
        >
          <div
            className="pf-l-stack pf-m-gutter"
          >
            <div
              className="pf-l-stack__item"
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <div
                className="makeStyles-container-1"
              >
                <div
                  aria-label="refresh-icon"
                  className="makeStyles-reloadButton-2"
                  id="refresh-icon"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="makeStyles-icon-4"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z"
                    />
                  </svg>
                </div>
                <div
                  aria-label="refetch-intervals"
                  className="pf-c-dropdown makeStyles-buttonTitle-3"
                  data-ouia-component-id="OUIA-Generated-Dropdown-1"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe={true}
                  id="refresh-dropdown"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label="refresh-label"
                    className="pf-c-dropdown__toggle pf-m-plain"
                    data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                    data-ouia-component-type="PF4/DropdownToggle"
                    data-ouia-safe={true}
                    disabled={false}
                    id="refresh-toggle"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    type="button"
                  >
                    <span
                      className="pf-c-dropdown__toggle-text"
                    >
                      Refresh every 10s
                    </span>
                    <span
                      className="pf-c-dropdown__toggle-icon"
                    >
                      <svg
                        aria-hidden={true}
                        aria-labelledby={null}
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style={
                          Object {
                            "verticalAlign": "-0.125em",
                          }
                        }
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                className="makeStyles-timestamp-5"
              >
                <span
                  aria-valuetext="Loading..."
                  className="pf-c-spinner pf-m-sm"
                  role="progressbar"
                >
                  <span
                    className="pf-c-spinner__clipper"
                  />
                  <span
                    className="pf-c-spinner__lead-ball"
                  />
                  <span
                    className="pf-c-spinner__tail-ball"
                  />
                </span>
                <p>
                  Last update: 
                  Invalid date
                </p>
              </div>
              <button
                aria-disabled={true}
                aria-label={null}
                className="pf-c-button pf-m-primary pf-m-aria-disabled"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                id="edit-policy"
                onClick={[Function]}
                onKeyPress={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                Edit policy
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
    <div
      style={
        Object {
          "borderTop": "thin solid #00000022",
        }
      }
    >
      <section
        className="pf-c-page__main-nav pf-m-light"
        style={
          Object {
            "paddingTop": 0,
          }
        }
      >
        <nav
          aria-label="Local"
          className="pf-c-nav pf-m-horizontal pf-m-tertiary"
          data-ouia-component-id="OUIA-Generated-Nav-tertiary-1"
          data-ouia-component-type="PF4/Nav"
          data-ouia-safe={true}
        >
          <button
            aria-label="Scroll left"
            className="pf-c-nav__scroll-button"
            disabled={false}
            onClick={[Function]}
            tabIndex={null}
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
              />
            </svg>
          </button>
          <ul
            className="pf-c-nav__list"
            onScroll={[Function]}
          >
            <li
              className="pf-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-1"
              data-ouia-component-type="PF4/NavItem"
              data-ouia-safe={true}
            >
              <a
                aria-current={null}
                className="pf-c-nav__link"
                onClick={[Function]}
                tabIndex={null}
              >
                Details
              </a>
            </li>
            <li
              className="pf-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-2"
              data-ouia-component-type="PF4/NavItem"
              data-ouia-safe={true}
            >
              <a
                aria-current="page"
                className="pf-c-nav__link pf-m-current"
                onClick={[Function]}
                tabIndex={null}
              >
                Status
              </a>
            </li>
          </ul>
          <button
            aria-label="Scroll right"
            className="pf-c-nav__scroll-button"
            disabled={false}
            onClick={[Function]}
            tabIndex={null}
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </button>
        </nav>
      </section>
    </div>
    <span
      aria-valuetext="Loading..."
      className="pf-c-spinner pf-m-xl patternfly-spinner"
      role="progressbar"
    >
      <span
        className="pf-c-spinner__clipper"
      />
      <span
        className="pf-c-spinner__lead-ball"
      />
      <span
        className="pf-c-spinner__tail-ball"
      />
    </span>
  </main>
</div>
`;
